pkg/queryfrontend/roundtrip.go

```diff
@@ -272,10 +272,20 @@ func newLabelsTripperware(
 	}, nil
 }
 
-// Don't go to response cache if StoreMatchers are set.
+// shouldCache controls what kind of Thanos request should be cached.
+// Don't cache:
+// * if StoreMatchers are set (since it is debug option).
+// * if Dedup is disabled (this is done only for debugging).
+// * Caching option disables cache.
```

Comment:
I'm not sure I get what this comment means?

Reply:
Yup this is confusing, are you referring to `Cache-Control` headers on the request?

Reply:
Yup